src/operators.jl

```diff
@@ -1,70 +1,39 @@
-        Base.Broadcast.promote_containertype(::Type{Array}, ::Type{ImageMeta}) = ImageMeta
-        Base.Broadcast.promote_containertype(::Type{ImageMeta}, ::Type{Array}) = ImageMeta
-        Base.Broadcast.promote_containertype(::Type{ImageMeta}, ct) = ImageMeta
-        Base.Broadcast.promote_containertype(ct, ::Type{ImageMeta}) = ImageMeta
-        Base.Broadcast.broadcast_indices(::Type{ImageMeta}, A) = indices(A)
-        function Base.Broadcast.broadcast_c(f, ::Type{ImageMeta}, As...)
-            T = _broadcast_eltype(f, As...)
-            shape = broadcast_indices(As...)
-            Mt = imagemeta(As...)
-            M = Mt[1]
-            if length(Mt) > 1
-                for i = 2:length(Mt)
-                    Mt[i] == M || return ambigop(Symbol(f))
-                end
-            end
-            broadcast!(f, shareproperties(M, similar(data(M), T, shape)), As...)
+# Specialize the low-level broadcasting machinery for ImageMeta.
+# This make it work for all operators, rather than specializing
+# each operator individually.
+Base.BroadcastStyle(::Type{<:ImageMeta}) = Broadcast.ArrayStyle{ImageMeta}()
+function Base.similar(bc::Broadcast.Broadcasted{Broadcast.ArrayStyle{ImageMeta}}, ::Type{ElType}) where ElType
+    Mt = imagemeta(bc.args...)
+    M = Mt[1]
+    if length(Mt) > 1
+        for i = 2:length(Mt)
+            Mt[i] == M || return ambigop(:Broadcast)
         end
     end
-    # Select all the ImageMeta arrays
-    @inline imagemeta(As...) = _imagemeta((), As...)
-    _imagemeta(out) = out
-    @inline _imagemeta(out, A::ImageMeta, As...) = _imagemeta((out..., A), As...)
-    @inline _imagemeta(out, A, As...) = _imagemeta(out, As...)
+    # Use the properties field of img to create the output
+    ImageMeta(similar(Array{ElType}, axes(bc)), M.properties)
 end
+# Select all the ImageMeta arrays
+@inline imagemeta(As...) = _imagemeta((), As...)
+_imagemeta(out) = out
+@inline _imagemeta(out, A::ImageMeta, As...) = _imagemeta((out..., A), As...)
+@inline _imagemeta(out, A, As...) = _imagemeta(out, As...)
 
-if dotops
-    import Base: .+, .-, .*, ./, .^, .<, .>, .==
-    batch1 = (:+, :.+, :-, :.-, :*, :.*)
-    batch2 = (:+, :.+, :-, :.-)
-else
-    batch1 = (:+, :-, :*)
-    batch2 = (:+, :-)
-end
+batch1 = (:+, :-, :*)
+batch2 = (:+, :-)
 
 (-)(img::ImageMeta) = shareproperties(img, -data(img))
 
+import Base.Broadcast: broadcasted, materialize
+
 for op in batch1
     @eval begin
-        ($op)(img::ImageMeta{Bool}, n::Bool) = shareproperties(img, ($op)(data(img), n))
-        ($op)(n::Bool, img::ImageMeta{Bool}) = shareproperties(img, ($op)(n, data(img)))
-        ($op)(img::ImageMeta, n::Number) = shareproperties(img, ($op)(data(img), n))
-        ($op)(n::Number, img::ImageMeta) = shareproperties(img, ($op)(n, data(img)))
+        broadcasted(::typeof($op),img::ImageMeta{Bool}, n::Bool) = shareproperties(img, materialize(broadcasted(($op),data(img), n)))
```

Comment:
Did you check whether we even need these anymore? (Maybe for ambiguity resolution?)

Reply:
Interesting suggestion.  If I comment out the whole section, the tests still pass. Is that sufficient evidence that this is needless specialization? (When I had it wrong during development, the tests failed, so this was definitely exercised.)

Reply:
If we have a test for `img .+ true` and it passes without them, we don't need it. Much better to delete.